File: easy/longest_common_prefix.py
def longest_common_prefix(strs):
    """
    DOCUMENTATION
    """
    if not strs or len(strs) == 0:
        return ""
    prefix = [""] * len(max(strs, key=len))
    for outdex, outer in enumerate(strs):
        for index, inner in enumerate(outer):
            if inner == strs[0][outdex]:
                prefix.insert(index, inner)
    return "".join(prefix)

File: easy/longest_common_prefix.go
package easy

func LongestCommonPrefix(strs []string) string {
	if len(strs) == 0 {
		return ""
	}
	for i := range len(strs[0]) {
		for _, word := range strs[1:] {
			if i >= len(word) || word[i] != strs[0][i] {
				return strs[0][:i]
			}
		}
	}
	return strs[0]
}
